### DISCUSSION TITLE
Windows Subsystem for Android (WSA) v2305.40000.6.0 has been released

### DISCUSSION THREAD

**Discussion by Almighty188** [LSPosed/MagiskOnWSALocal#600] [General]
Windows Subsystem for Android (WSA) v2305.40000.6.0 has been released

Current version(s) of Windows Subsystem for Android (WSA):
Fast 2305.40000.6.0
Slow 2305.40000.6.0
Release Preview 2305.40000.6.0
Retail 2304.40000.10.0

**Comment by supermansaga** [comment]
So Release Preview Channel is slower than Beta and Dev Channel? At this point, I presume Android 13 beta is embedded? But how can I tell? Thanks.

**Reply by Almighty188** [reply]
Release Preview, Beta and Dev channels have the exact same version(s) it seems.  Retail is the one that is slower than the rest.   No idea on your last question I think Android 13 was introduced as one of the versions, not sure what version number though. 

Look at https://learn.microsoft.com/en-us/windows/android/wsa/release-notes and you will see that Android 13 is in Build 2211.40000.11.0 and later.

**Reply by fromhgbwithluv** [reply]
Android 13 have been embeded

**Reply by supermansaga** [reply]
Thank you all, but it's really hard to feel its presence. Where else could I see it's really Android 13, other than by referring to 2305 itself? Does it have a graphical interface indicating so? Thanks.

**Comment by Almighty188** [comment]
Well, remember WSA is a emulator, and I could be wrong but all it really does is it allows you to run Android apps and is not actually Android OS ported to Windows so it will support anything that uses Android 13 and before as I remember it converts the code so Windows understands it using something called Intel Bridge Technology as mentioned here:
https://learn.microsoft.com/en-us/windows/android/wsa/

https://www.theverge.com/2021/6/24/22549303/windows-11-intel-bridge-android-apps-amd-arm-processors

So WSA is not Android running under Windows but more to allow Android apps to be able to run under Windows as Intel Bridge technology compiles the code in real-time.    So the only thing you can do is really find Android apps that shows the Android OS version and see what it says like this:
https://play.google.com/store/apps/details?id=com.octabeans.osversioninfoforandroid&hl=en_US&gl=US
or this which works as you can see:
https://play.google.com/store/apps/details?id=com.catchy.tools.myphoneinfo.rp&hl=en_US&gl=US
<img width="1264" alt="image" src="https://github.com/LSPosed/MagiskOnWSALocal/assets/78993393/1b66bbb0-7fd6-49ab-9f74-7886de637f31">

or this:
https://play.google.com/store/apps/details?id=flar2.devcheck&hl=en_US&gl=US

If you don't believe what Microsoft tells you since it just has support for Android 13 functions but it is not Android 13 running under Windows as you don't have the setings -> About. screen like you do on a real Android device.  It's no different than Samsung telling you, you are running Android 13 but even uname -a will only show you the Linux kernel version which is different than the actual Android version.  

![image](https://github.com/LSPosed/MagiskOnWSALocal/assets/78993393/9acc0b0c-d22d-4c82-9257-1962c6e33d5a)

It's hard to feel the presence because remember this is a software device and not a physical device, same reason I hate running other OSes as a virtual machine as it doesn't feel real.  And Microsoft or anyone else for that matter can configure the answer in the OS so it can answer as Android v99 but that doesn't mean that's the real Android version used as it could be v7 for all we know.  I can make WSA itself say something other than 2305.40000.6.0 if I wanted to simply by hex editing the binaries and modifying the 2305.40000.6.0 string.

**Reply by supermansaga** [reply]
Many thanks for the nuance. I highly appreciate it. Best regards,

**Reply by Almighty188** [reply]
No problem, curiosity kills... LOL

**Reply by supermansaga** [reply]
Absolutely, didn't go to bed until 6am. It surely killed my health

**Reply by Almighty188** [reply]
I haven't slept for the last 46 hours.  But if you find an app that does not have ads that shows the Android version, please post it in this thread...

**Reply by Almighty188** [reply]
Seems this one is the better app to display system information...

https://play.google.com/store/apps/details?id=deviceinfo.systeminfo.cpu.device.system&hl=en_US&gl=US
![image](https://github.com/LSPosed/MagiskOnWSALocal/assets/78993393/651bdb48-3bde-4122-bf58-108d943accb6)
